include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,618 @@
+     */
+    constexpr iterator_type begin() noexcept
+    {
+        return iterator_type{std::ranges::begin(src_range), std::ranges::begin(src_range), std::ranges::end(src_range)};
+    }
+
+    //!\copydoc begin()
+    constexpr iterator_type begin() const noexcept
+    {
+        return iterator_type{std::ranges::begin(src_range), std::ranges::begin(src_range), std::ranges::end(src_range)};
+    }
+
+    //!\copydoc begin()
+    constexpr iterator_type cbegin() const noexcept
+    {
+        return begin();
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the range.
+     * \returns Iterator to the end.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr iterator_type end() noexcept
+    {
+        return iterator_type{back_iterator, std::ranges::begin(src_range), std::ranges::end(src_range)};
+    }
+
+    //!\copydoc end()
+    constexpr iterator_type end() const noexcept
+    {
+        return iterator_type{back_iterator, std::ranges::begin(src_range), std::ranges::end(src_range)};
+    }
+
+    //!\copydoc end()
+    constexpr iterator_type cend() const noexcept
+    {
+        return end();
+    }
+    //!\}
+
+private:
+
+    //!\brief The underling range.
+    source_range_type src_range{};
+    //!\brief The cached iterator pointing to the last element of the source range.
+    std::ranges::iterator_t<source_range_type> back_iterator{};
+};
+
+/*!\name Type deduction guides
+ * \{
+ */
+
+//!\brief Deduces the correct template type from a view.
+template <std::ranges::View other_range_t>
+pairwise_combine_view(other_range_t range) -> pairwise_combine_view<other_range_t>;
+
+//!\brief Deduces the correct template type from a non-view lvalue range by wrapping the range in seqan3::view::all.
+template <std::ranges::ViewableRange other_range_t>
+pairwise_combine_view(other_range_t && range) ->
+    pairwise_combine_view<decltype(view::all(std::declval<other_range_t &&>()))>;
+//!\}
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief             A view adapter that generates all pairwise combinations of the elements of the underlying range.
+ * \tparam urng_t     The type of the range being processed. See below for requirements.
+ * \param[in] urange  The range being processed.
+ * \returns           A view over all pairwise combinations of the elements of the underlying range.
+ *                    See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view generates two-element tuples representing all unique combinations of the elements of the underlying range
+ * (the order of the elements does not matter). If the underlying range has less than two elements the returned range is
```

Comment:
`(the order of elements is undefined)` ?
"does not matter" is very colloquial. 💅

Reply:
Then wikipedia is colloquial 😱